lib/repl.js

```diff
@@ -70,6 +70,88 @@ const BLOCK_SCOPED_ERROR = 'Block-scoped declarations (let, ' +
     'const, function, class) not yet supported outside strict mode';
 
 
+class LineParser {
+
+  constructor() {
+    this.reset();
+  }
+
+  reset() {
+    this._literal = null;
+    this.shouldFail = false;
+    this.blockComment = false;
+  }
+
+  parseLine(line) {
+    var previous = null;
+    this.shouldFail = false;
+    const wasWithinStrLiteral = this._literal !== null;
+
+    for (const current of line) {
```

Comment:
It works and has the advantage to handle unicode correctly:
![image](https://cloud.githubusercontent.com/assets/2352663/10741088/2d5ad6e0-7c26-11e5-8ec8-e2bc6e1112d9.png)